src/mongo/db/pipeline/expression.cpp

```diff
@@ -1729,9 +1729,18 @@ intrusive_ptr<ExpressionObject> ExpressionObject::parse(
 }
 
 intrusive_ptr<Expression> ExpressionObject::optimize() {
+    size_t countConstants = 0;
     for (auto&& pair : _expressions) {
+        // Check if Expression Constant
```

Comment:
Use full sentences in comments. This one seems like it's not adding much that the next line doesn't already say, so I'd actually suggest removing it.